src/terminator/mod.rs

```diff
@@ -138,6 +131,69 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         Ok(())
     }
 
+    /// Decides whether it is okay to call the method with signature `real_sig` using signature `sig`
```

Comment:
That's almost literally impossible. The only thing we can do is track down all publicly documented ABIs, encode their rules, and apply the rules of *all* ABIs at once, comparing the result.
You *could* try to approximate a common core, but it only makes sense as the intersection of all supported ABIs. It's also a bit risky to specify due to forwards compatibility.